src/main/java/org/wildfly/security/http/HttpServerRequest.java

```diff
@@ -63,6 +64,13 @@
     SSLSession getSSLSession();
 
     /**
+     * Get the peer certificates established on the connection.
+     *
+     * @return the peer certificates established on the connection or {@code null} if none available.
+     */
+    Certificate[] getPeerCertificates();
```

Comment:
I did think about that as well but it seems the history of SSLSession has been to move from X509Certificate to Certificate